### DISCUSSION TITLE
How to implement a custom filter (not refiner)

### DISCUSSION THREAD

**Discussion by pettersh** [microsoft-search/pnp-modern-search#1520] [Q&A] [answered]
We need guidance on how to implement a custom filter to show only results from "trusted sources". 

We're porting a custom angular solution built for SharePoint 2013 to PnP Modern Search v4 / SPO and would like to continue with a "toggle" or similar control like screenshot. 

![image](https://user-images.githubusercontent.com/2064929/141786646-e4509c49-c20f-4239-8709-31ea805e374b.png)

We cannot use search refiners due to the complex requirements, and would like to to apply a query template dynamically when the user toggles or clicks the control. @wobba you know which solution this is from your past.

We already have a results page with several verticals with connected results webparts, and would ideally like to update the query template of current webpart by code. Alternatively we would like to enable/disable results webpart.

![image](https://user-images.githubusercontent.com/2064929/141788485-99d263c0-47cc-4fb3-a8b3-4aa42497d15f.png)

Here's the current implementation/query template

```
{searchboxquery} 
(
	(
		managed property:"value1" OR 
		managed property:"value2" OR 
		managed property:"value3" OR 
		managed property:"value4" OR 
		managed property:"value5" OR 
		managed property:"value6" 
	)   
	declaredrecorddate > 1900-01-01 
) OR 
(
	Path:"https://tenant.sharepoint.com/sites/dms/Lists/Published Revisions/"      
	IsDocument=1
)
```

**Comment by wobba** [comment, marked as answer]
If you replicate the filter web part, but replace the internal logic to pass in fql with your restriction when clicked. Or have some button add a URL param which you pick up in the query template when clicked.
